image/gen/src/generator.rs

```diff
@@ -115,62 +140,129 @@ impl<Crypto: ImageGeneratorCrypto> ImageGenerator<Crypto> {
         Ok(image)
     }
 
+    fn mldsa_sign(
+        &self,
+        digest: &ImageDigest512,
+        priv_key: &ImageMldsaPrivKey,
+    ) -> anyhow::Result<ImageMldsaSignature> {
+        pub fn from_hw_format<const NUM_BYTES: usize>(value: &[u32]) -> [u8; NUM_BYTES] {
+            let mut result = [0u8; NUM_BYTES];
+            for (i, &item) in value.iter().enumerate() {
+                let bytes = item.to_be_bytes();
+                let start = i * 4;
+                if start + 4 <= NUM_BYTES {
+                    result[start..start + 4].copy_from_slice(&bytes);
+                } else {
+                    break; // Prevent overflow if the input slice is larger than NUM_BYTES / 4
+                }
+            }
+            result
+        }
+
+        let mut rng = StdRng::from_seed([0u8; 32]);
```

Comment:
Okay, I looked it up, and this should be fine.

It's not the true deterministic signature version -- that one we need to pass an RNG that always returns 0, weirdly, because of how the `fips204` crate work.

I'll file an issue with the crate to add a method to support a non-hacky way to do deterministic signatures: https://github.com/integritychain/fips204/issues/8